### ISSUE TITLE
bug: HFBackend ignores explicit head_dim and rejects valid model KV tensors

### ISSUE THREAD

**Issue by umran666** [umran666/UniqKache#70]
### Bug and location

[src/uniqkache/models/hf_backend.py:99-113](https://github.com/umran666/UniqKache/blob/0b878a1acc3f2a84ea0ff16194456b8ed10191e8/src/uniqkache/models/hf_backend.py#L99-L113) always calculates head dimension as `hidden_size // num_attention_heads`, ignoring an explicit `config.head_dim`. Valid HF architectures/configurations may use a different attention projection dimension.

The resulting UniqKache cache has the wrong shape and rejects otherwise valid model outputs during mirroring.

### Reproduction

No model download is needed:

```python
import torch
from transformers import LlamaConfig, LlamaForCausalLM
from uniqkache import KVCache
from uniqkache.models.hf_backend import HFBackend

cfg = LlamaConfig(
    vocab_size=32, hidden_size=32, intermediate_size=64,
    num_hidden_layers=1, num_attention_heads=4,
    num_key_value_heads=2, head_dim=16,
)
model = LlamaForCausalLM(cfg).eval()
prompt = torch.tensor([[1, 2, 3]])
print(model(prompt).logits.shape)
backend = HFBackend(model, identifier="local-random-llama",
                    weights_are_random=True)
print(backend.config["head_dim"])
backend.forward(prompt, KVCache(backend.cache_config()))
```

Observed with Transformers 5.16.1: native logits shape `[1, 3, 32]`; adapter head dimension `8`; adapter forward raises `layer 0 expects head_dim 8, received 16`.

### Impact and expected behavior

The advertised full-cache HF execution path fails for valid models whose attention head width is not the hidden-size quotient. Shape-dependent memory accounting is also initialized incorrectly.

Use the architecture's explicit head dimension when present and derive it only as a fallback. Add a locally initialized model regression with unequal explicit and derived dimensions. This does not depend on eviction support or the HF cache-reset issue.

### Verification

Reviewed commit: `0b878a1acc3f2a84ea0ff16194456b8ed10191e8`. Reproduced on Windows with Python 3.10 and PyTorch 2.6.0+cu124; CPU unless otherwise specified. Synthetic-model results below are correctness diagnostics, not language-model quality claims.
